internal/util/util.go

```diff
@@ -110,3 +115,17 @@ func IsolateContextFromParentContext(ctx context.Context) (context.Context, cont
 	ctx = context.WithoutCancel(ctx)
 	return context.WithCancel(ctx)
 }
+
+// SecondsToTimeDuration ensure no int64 overflow while converting secs to
+// time.Duration.
+func SecondsToTimeDuration(secs int64) time.Duration {
+	if secs < 0 {
+		return MaxTimeDuration
+	}
+
+	if secs > (math.MaxInt64 / int64(time.Second)) {
```

Comment:
i think we should not return MaxTimeDuration here and rather handle this case (throw error) in parsing the config itself.